pkg/instrumentors/bpf/net/http/server/bpf/probe.bpf.c

```diff
@@ -50,6 +69,84 @@ volatile const u64 method_ptr_pos;
 volatile const u64 url_ptr_pos;
 volatile const u64 path_ptr_pos;
 volatile const u64 ctx_ptr_pos;
+volatile const u64 headers_ptr_pos;
+
+static __always_inline struct span_context* extract_context_from_req_headers(void *headers_ptr_ptr) {
+    void *headers_ptr;
+    long res;
+    res = bpf_probe_read(&headers_ptr, sizeof(headers_ptr), headers_ptr_ptr);
+    if (res < 0) {
+        return NULL;
+    }
+    u64 headers_count = 0;
+    res = bpf_probe_read(&headers_count, sizeof(headers_count), headers_ptr);
+    if(res < 0) {
+        return NULL;
+    }
+    if (headers_count == 0) {
+        return NULL;
+    }
+    unsigned char log_2_bucket_count;
+    res = bpf_probe_read(&log_2_bucket_count, sizeof(log_2_bucket_count), headers_ptr + 9);
+    if(res < 0) {
+        return NULL;
+    }
+    u64 bucket_count = 1 << log_2_bucket_count;
+    void *header_buckets;
+    res = bpf_probe_read(&header_buckets, sizeof(header_buckets), headers_ptr + 16);
+    if(res < 0) {
+        return NULL;
+    }
+    u32 map_id = 0;
+    struct map_bucket *map_value = bpf_map_lookup_elem(&golang_mapbucket_storage_map, &map_id);
+    if (!map_value) {
+        return NULL;
+    }
+
+    for (u64 j=0; j<MAX_BUCKETS; j++) {
+        if(j >= bucket_count) {
+            break;
+        }
+        res = bpf_probe_read(map_value, sizeof(struct map_bucket), header_buckets + (j * sizeof(struct map_bucket)));
+        if(res < 0) {
+            continue;
+        }
+        for (u64 i=0; i<8; i++) {
```

Comment:
Would not make sense to extract `8` as constant? and use it also in struct

Reply:
@pdelewski I added a specific TODO when consolidating the HTTP probes to ensure const values are used for max loop iterations. Resolving this comment so we can get this merged then refactor.